cfme/control/explorer/policy.py

```diff
@@ -0,0 +1,299 @@
+
+from . import ControlExplorerView
+from utils.appliance import Navigatable
+from utils.update import Updateable
+
+
+def policies_all_view_class_constructor(policy_object):
+
+    def is_displayed(self):
+        return (
+            ControlExplorerView.is_displayed and
+            self.title.text == "All {} {} Policies".format(
+                policy_object.PRETTY,
+                policy_object.TYPE
+            )
+        )
+
+    policies_all_views_class = type("PoliciesAllView", (ControlExplorerView, ), {
+        "title": Text("#explorer_title_text"),
+        "is_displayed": property(is_displayed)
+    })
+
+    return policies_all_views_class
+
+
+class PolicyFormCommon(ControlExplorerView):
+
+    description = Input(name="description")
+    active = Checkbox("active")
+    scope = BootstrapSelect("chosen_typ")
+    notes = TextInput(name="notes")
+
+    cancel_button = Button('Cancel')
+
+
+class NewPolicyView(PolicyFormCommon):
+    title = Text("#explorer_title_text")
+
+    add_button = Button("Add")
+
+    @property
+    def is_displayed(self):
+        return all((
+            super(NewPolicyView, self).is_displayed,
+            self.title.text == "Adding a new {} {} Policy".format(
+                self.context["object"].PRETTY, self.context["object"].TYPE),
+            self.policies.is_opened,
+            self.policies.tree.currently_selected == ["All {} {} Policies".format(
+                self.context["object"].PRETTY, self.context["object"].TYPE)]
+        ))
+
+
+class EditPolicyView(PolicyFormCommon):
+    title = Text("#explorer_title_text")
+
+    save_button = Button("Save")
+    reset_button = Button("Reset")
+
+    @property
+    def is_displayed(self):
+        return all((
+            super(EditPolicyView, self).is_displayed,
+            self.title.text == 'Editing Action "{}"'.format(self.context["object"].description),
+            self.policies.is_opened,
+            self.policies.tree.currently_selected == ["All Actions"]
+        ))
+
+
+class PolicyDetailsView(ControlExplorerView):
+    title = Text("#explorer_title_text")
+
+    type = SummaryFormItem("Basic Information", "Action Type")
+    analysis_profile = SummaryFormItem("Analysis Profile", "Assigned Analysis Profile")
+
+    @property
+    def is_displayed(self):
+        return all((
+            super(PolicyDetailsView, self).is_displayed,
+            self.title.text == '{} {} Policy "{}"'.format(self.context["object"].PRETTY,
+                self.context["object"].TYPE, self.context["object"].description),
+            self.policies.is_opened,
+            self.policies.tree.currently_selected == [
+                "All Policies",
+                "{} Policies".format(self.context["object"].TYPE),
+                "{} {} Policies".format(self.context["object"].TREE_NODE,
+                    self.context["object"].TYPE),
+                self.context["object"].description
+            ]
+        ))
```

Comment:
Problem of `all` is that it does not short-circuit when one of the conditions is false, this would slow things down I guess.